src/dune_pkg/resolved_package.ml

```diff
@@ -84,10 +84,10 @@ let scan_files_entries path =
          ])
 ;;
 
-let dune_package loc opam_file opam_package =
+let dune_package ~dune_build loc opam_file opam_package =
```

Comment:
This code doesn't seem to make sense to me. `dune_package` should return a `dune_package` but with `~dune_build:false` it doesn't? It needs to be renamed at least but possibly a bit of code needs to be refactored (e.g. by passing around the `build_type` type I am suggesting) and renaming.

Reply:
What do you think about keeping the function, adding a new one with the parameter at false? Maybe it would make more sense?

Reply:
I think the way you wrote it at the moment looks good to me.